### ISSUE TITLE
HOGDescriptor

### ISSUE THREAD

**Issue by pefi9** [bytedeco/javacpp-presets#109]
Hi,
I can't get HOG descriptors, respectively it's all zeros.
```java
 opencv_objdetect.HOGDescriptor hogDesc = new opencv_objdetect.HOGDescriptor(
            new Size(64, 128)
            , new Size(16, 16)
            , new Size(8, 8)
            , new Size(8, 8)
            , 9);

 float[] desc = new float[(int)hogDesc.getDescriptorSize()];
  Mat temp = cvarrToMat(img);
  hogDesc.compute(temp, desc);

```

**Comment by saudet** [comment]
Have you tried to do the same thing as shown in this OpenCV sample?
https://github.com/Itseez/opencv/blob/3.0.0/samples/tapi/hog.cpp

**Comment by pefi9** [comment]
My understanding of this sample is that it`s for detecting people with pre-trained  SVM classificator: hog.setSVMDetector( HOGDescriptor::getDaimlerPeopleDetector() );

I`m trying to create simple OCR where HOG descriptor would provide me gradients of the different parts of an image and svm would be trained based on these descriptors. 
More like the 2nd answer here: http://answers.opencv.org/question/10374/how-to-training-hog-and-use-my-hogdescriptor/

**Comment by saudet** [comment]
Ok, so would you have some code in C++ that is known to be working? And if you cannot find anything, please ask the authors of `HOGDescriptor` for usage guidelines. It's not like I'm supposed to know how to use that undocumented API.

**Comment by pefi9** [comment]
Sure, thanks.

**Comment by saudet** [comment]
I've tries this here and it returns something else than zeros:
```java
    Mat image = imread("/home/saudet/projects/bytedeco/javacv/samples/pic1.png");
    HOGDescriptor hog = new HOGDescriptor();
    FloatPointer descriptors = new FloatPointer();
    hog.compute(image, descriptors);
```
But `descriptors.limit()` is much larger than `hog.getDescriptorSize()`, so I guess you're only getting the beginning of the data. You need to specify a larger array if you want to use `float[]` instead of `FloatPointer`.

**Comment by pefi9** [comment]
I see, thanks a lot.

**Comment by pefi9** [comment]
I tried it with "FloatPointer" instead of float[]. The size is just the same however FloatPointer is populated with number after calling hogDescriptor.comput while float[] is empty (all zeros).

Anyway, FloatPointer works - thanks for the hind.

**Comment by saudet** [comment]
@pefi9 Yes, it seems there's a problem there... I'll have to look into that.

**Comment by pefi9** [comment]
Hi @saudet,
how can I load trained svm model? I saved the model by using svm.save(path), but I can`t figure out how to load it (something like svm.load(path)?).

**Comment by saudet** [comment]
What do they do in the C++ samples for that?

**Comment by pefi9** [comment]
C++: void CvStatModel::load(const char* filename, const char* name=0 )
Taken from:
http://docs.opencv.org/2.4/modules/ml/doc/statistical_models.html 

Example:
CvSVM *SVM = new CvSVM;
SVM->load("SVM.xml");
SVM->predict(DataRow);
http://answers.opencv.org/question/13418/machine-learning-saveload-problem/

**Comment by saudet** [comment]
That's for OpenCV 2.4. Any examples for OpenCV 3.0?

**Comment by pefi9** [comment]
http://docs.opencv.org/3.0-beta/modules/ml/doc/statistical_models.html#statmodel-load

**Comment by saudet** [comment]
I see, they've moved the `load()` methods to function templates. I've added something to the commit above. Let me know if this satisfactory! I've also fixed the problem with the `HOGDescriptor.compute()` and `int[]` in the other commit. Thanks for reporting!

**Comment by taobupt** [comment]
there is still no function **load()**  in SVM, so what should I do if I want to load a classifier file from a path. please give a hand

**Comment by saudet** [comment]
@taobupt It's all here: https://github.com/bytedeco/javacpp-presets/blob/master/opencv/src/main/java/org/bytedeco/javacpp/helper/opencv_ml.java#L48

**Comment by taobupt** [comment]
there is no opencv_ml.java in *javacv-1.1-bin.zip (140 MB), *which means
that I have to compile the source code by myself? you can add it to the
javacv-1.1-bin.zip if you have some free time. Thanks for help.

On Sun, Apr 3, 2016 at 6:48 PM, Samuel Audet <notifications@github.com>
wrote:

> @taobupt <https://github.com/taobupt> It's all here:
> https://github.com/bytedeco/javacpp-presets/blob/master/opencv/src/main/java/org/bytedeco/javacpp/helper/opencv_ml.java#L48
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/bytedeco/javacpp-presets/issues/109#issuecomment-204940706>
>



-- 
Email：wangtaoo665@gmail.com
Phone：8613261706096

**Comment by saudet** [comment]
Please follow the build instructions here, if _you_ have some time:
https://github.com/bytedeco/javacpp-presets/#build-instructions

**Comment by saudet** [comment]
This has all been fixed and included in the just released version 1.2. Thanks for reporting all this guys!
